src/Builder.php

```diff
@@ -396,6 +397,12 @@ private function buildUrl(string $url, bool $ignoreContentNotFoundErrors = false
         $output = ob_get_clean();
         $content = $response->getContent() ?: $output;
 
+        if ($this->hasNoIndexHeader($response) && !$this->hasNoIndexTag($content)) {
+            $this->logger->warning('Url "{url}" contains a "x-robots-tag: noindex" header that will be lost by going static. Consider using the HTML tag "<meta name="robots" content="noindex" />" instead.', [
```

Comment:
Since the progress bar (used in normal verbosity and interactive mode) may "eat" the first logs by redrawing over, the user might miss the warnings if there are only a few of them.
Note this is not an issue in CI (non-interactive), or whenever using another verbosity level (`-v` is enough). But I've 1 or 2 idea(s) for gathering such warnings to be displayed at the end of the command execution with the progress bar.